sdk/python/tests/unit/permissions/auth/server/test_auth_registry_server.py

```diff
@@ -65,33 +76,149 @@ def remote_feature_store(server_port, feature_store):
 
 
 def test_registry_apis(
-    auth_config, temp_dir, server_port, start_registry_server, remote_feature_store
+    auth_config,
+    temp_dir,
+    server_port,
+    start_registry_server,
+    remote_feature_store,
+    applied_permissions,
 ):
     print(f"Runnning for\n:{auth_config}")
-    _test_list_entities(remote_feature_store)
-    _test_list_fvs(remote_feature_store)
+    permissions = _test_list_permissions(remote_feature_store, applied_permissions)
+    _test_list_entities(remote_feature_store, applied_permissions)
+    _test_list_fvs(remote_feature_store, applied_permissions)
+    _test_get_historical_features(permissions, remote_feature_store)
 
 
-def _test_list_entities(fs: FeatureStore):
-    entities = fs.list_entities()
+def _test_get_historical_features(
+    permissions: list[Permission], client_fs: FeatureStore
+):
+    if _permissions_exist_in_permission_list(
```

Comment:
can we move this check outside of the function and let the test function decide whether to invoke this validation step or not?